vm_insnhelper.c

```diff
@@ -5846,6 +5846,33 @@ vm_ic_update(const rb_iseq_t *iseq, IC ic, VALUE val, const VALUE *reg_ep, const
     rb_rjit_constant_ic_update(iseq, ic, pos);
 }
 
+static VALUE
+vm_opt_getconstant_path(rb_execution_context_t *ec, rb_control_frame_t *const reg_cfp, IC ic)
+{
+    VALUE val;
+    const ID *segments = ic->segments;
+    struct iseq_inline_constant_cache_entry *ice = ic->entry;
+    if (ice && vm_ic_hit_p(ice, GET_EP())) {
+        val = ice->value;
+
+        VM_ASSERT(val == vm_get_ev_const_chain(ec, segments));
+    } else {
+        ruby_vm_constant_cache_misses++;
+        val = vm_get_ev_const_chain(ec, segments);
+        vm_ic_track_const_chain(GET_CFP(), ic, segments);
+        // Because leaf=false, we need to undo the PC increment to get the address to this instruction
+        // INSN_ATTR(width) == 2
+        vm_ic_update(GET_ISEQ(), ic, val, GET_EP(), GET_PC() - 2);
+    }
+    return val;
+}
+
+VALUE
+rb_vm_opt_getconstant_path(rb_execution_context_t *ec, rb_control_frame_t *const reg_cfp, IC ic)
```

Comment:
Nit: maybe have just the non-static `rb_` variant for code size in this case? Up to you.

Reply:
It was a deliberate choice for potentially saving a function call overhead since `static` could be inlined whereas non-`static` functions are never inlined. Looking at the code generated by GCC 11.4, however, `vm_opt_getconstant_path` was compiled as a separate function. So I removed the unneeded wrapper. `rb_vm_opt_getconstant_path` had only `endbr64` and `jmp` (+ `nop`) and the code size impact wasn't too bad, but it'd be nice to remove a `jmp` instruction for JIT.